### PR TITLE
Add deployment guide

### PR DESCRIPTION
**Pull request by tddough98**
This adds a markdown file to walk through creating and configuring the cloud resources to put DeepCell Label online. As I'll be leaving next month, it's intended to serve as a resource for future work on the project to make sure it can be redeployed and updated. As we discover new required configuration and switch the resources we use, we should update this guide with the up to date steps for deployment.

### CODE REVIEW

File: DEPLOYMENT.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+Install the Elastic Beanstalk command line tools following [these instructions](https://docs.aws.amazon.com/elasticbeanstalk/latest/dg/eb-cli3-install.html). You can check that they are installed by running `eb --version` on the command line.
+
+If using Elastic Beanstalk for the first time in the repository, run `eb init` from the root of the repository to initialize Elastic Beanstalk.
+
+use `eb create`. creating an Elastic Beanstalk environment for the first time,
+
+To use existing environment, run `eb use ENVIRONMENT_NAME` to set the environment for the current git branch. Each time you switch git branches, make sure to set the environment with `eb use`.
+
+Once the environment is set, run `eb status`. Look for CNAME to see the instance's endpoint URL. You can also find this URL on the [AWS Console](https://us-east-2.console.aws.amazon.com/elasticbeanstalk/home). Make a note of this URL to use with CloudFront.
+
```
Comment (msschwartz21) [inline DEPLOYMENT.md:64]:
```suggestion

Before deploying the instance, confirm that you have set the following values in the `.env` file:
- `AWS_ACCESS_KEY_ID`
- `AWS_SECRET_ACCESS_KEY`
- `S3_BUCKET` - this is the bucket where you will store files uploaded to DCL 
- `SQLALCHEMY_DATABASE_URI`

```

**Review by msschwartz21 — CHANGES_REQUESTED** [review]
Looks great! Everything went pretty well with my run through. I left suggestions in places that I thought needed additional clarification.

File: DEPLOYMENT.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+- Invalidate the CloudFront cache for `/` and `/index.html` so the updated assets are served
+
+## Appendix: using the AWS RDS database
+
+A MySQL database is set up with RDS on AWS. It's not part of the deployment process, other than providing credentials to the backend to connect to it.
+
+In case the database requires changes, access it on the command like with
+
+```
+mysql --host=deepcell-label-db-1.cbwvcgkyjfot.us-east-2.rds.amazonaws.com --user=deepcelladmin --password -P 3306
```
[the remark this answers is not in the document]
Comment (tddough98) [reply DEPLOYMENT.md:138]: This information is not sensitive, but it's not a bad idea to take it offline. It reveals the endpoint for the database and the username, but not the password. I will put the complete command in the admin slack channel and put placeholders here instead

**Review by msschwartz21 — APPROVED** [review]

**Comment by WeihaoGe1009** [comment]
To who this may concern,

I was trying to follow the deployment guide on an AWS EC2 instance and failed at a very early stage. 
besides installing prerequisits for mysql-client, node.js, npm, and yarn, now i'm at the very first step following https://github.com/vanvalenlab/deepcell-label/blob/main/documentation/DEPLOYMENT.md, section "Host a static site with an S3 bucket". I have created a bucket and updated the policy, and then try to build the front end with yarn. Then, I was kept with the error message:

../angle/src/libANGLE/HandleAllocator.cpp: In constructor ‘gl::HandleAllocator::HandleAllocator()’:
../angle/src/libANGLE/HandleAllocator.cpp:29:52: error: ‘numeric_limits’ is not a member of ‘std’
   29 |     mUnallocatedList.push_back(HandleRange(1, std::numeric_limits<GLuint>::max()));
      |                                                    ^~~~~~~~~~~~~~
../angle/src/libANGLE/HandleAllocator.cpp:29:76: error: ‘::max’ has not been declared; did you mean ‘std::max’?
   29 |     mUnallocatedList.push_back(HandleRange(1, std::numeric_limits<GLuint>::max()));
      |                                                                            ^~~
      |                                                                            std::max
In file included from /usr/include/c++/11/algorithm:62,
                 from ../angle/src/libANGLE/HandleAllocator.cpp:12:
/usr/include/c++/11/bits/stl_algo.h:3467:5: note: ‘std::max’ declared here
 3467 |     max(initializer_list<_Tp> __l, _Compare __comp)
      |     ^~~
../angle/src/libANGLE/HandleAllocator.cpp:29:43: error: expected primary-expression before ‘(’ token
   29 |     mUnallocatedList.push_back(HandleRange(1, std::numeric_limits<GLuint>::max()));
      |                                           ^
../angle/src/libANGLE/HandleAllocator.cpp:29:52: error: ‘numeric_limits’ is not a member of ‘std’
   29 |     mUnallocatedList.push_back(HandleRange(1, std::numeric_limits<GLuint>::max()));
      |                                                    ^~~~~~~~~~~~~~
../angle/src/libANGLE/HandleAllocator.cpp:29:73: error: expected primary-expression before ‘>’ token
   29 |     mUnallocatedList.push_back(HandleRange(1, std::numeric_limits<GLuint>::max()));
      |                                                                         ^
../angle/src/libANGLE/HandleAllocator.cpp:29:76: error: ‘::max’ has not been declared; did you mean ‘std::max’?
   29 |     mUnallocatedList.push_back(HandleRange(1, std::numeric_limits<GLuint>::max()));
      |                                                                            ^~~
      |                                                                            std::max
In file included from /usr/include/c++/11/algorithm:62,
                 from ../angle/src/libANGLE/HandleAllocator.cpp:12:
/usr/include/c++/11/bits/stl_algo.h:3467:5: note: ‘std::max’ declared here
 3467 |     max(initializer_list<_Tp> __l, _Compare __comp)
      |     ^~~
make: *** [angle/src/libANGLE.target.mk:241: Release/obj.target/libANGLE/angle/src/libANGLE/HandleAllocator.o] Error 1
make: Leaving directory '/home/ec2-user/deepcell-label/frontend/node_modules/gl/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/home/ec2-user/deepcell-label/frontend/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (node:events:517:28)
gyp ERR! stack     at ChildProcess._handle.onexit (node:internal/child_process:292:12)
gyp ERR! System Linux 6.1.119-129.201.amzn2023.x86_64
gyp ERR! command "/home/ec2-user/.nvm/versions/node/v18.20.5/bin/node" "/home/ec2-user/deepcell-label/frontend/node_modules/gl/node_modules/.bin/node-gyp" "rebuild"
gyp ERR! cwd /home/ec2-user/deepcell-label/frontend/node_modules/gl

I tried different ways to patch the file HandleAllocator.cpp with '#include <limits>' but it always got reset during the installation. Sometimes I was able to fix it after node-gyp rebuild, but never able to get through the yarn command, i.e. the very first command on the page. Could you help me to deploy DeepCell Label on AWS please?

**Pull request merged**